libhpcalcs/src/prime_cmd.c

```diff
@@ -360,47 +364,124 @@ HPEXPORT int HPCALL calc_prime_r_recv_screen(calc_handle * handle, calc_screensh
     return res;
 }
 
-// Seems to be made of a series of CMD_PRIME_RECV_FILE.
+HPEXPORT int HPCALL calc_prime_s_enable_new_protocol(calc_handle * handle) {
+    int res;
+    res = prime_send_new_protocol_init(handle);
+    if (!res) {
+        handle->protocol_version = 1;
+    }
+    return res;
+}
+
+HPEXPORT int HPCALL calc_prime_r_enable_new_protocol(calc_handle * handle) {
+    int res = 0;
+    if (handle == NULL) {
+        res = ERR_INVALID_HANDLE;
+        hpcalcs_error("%s: handle is NULL", __FUNCTION__);
+    }
+    return res;
+}
+
+HPEXPORT int HPCALL calc_prime_s_disable_new_protocol(calc_handle * handle) {
+    int res;
+    // It looks like sending a status check using the old protocol drops the Prime
+    // out of the new protocol mode.
+    res = calc_prime_s_check_ready(handle);
+    if (res) {
+        return res;
+    }
+    return res;
+}
+
+HPEXPORT int HPCALL calc_prime_r_disable_new_protocol(calc_handle * handle) {
+    int res;
+    if (handle != NULL) {
+        // While the prime is in thew new protocol mode, it sends us packages
+        // starting with 0xFE. We need to eliminate those now (and also wait for
```

Comment:
It actually sends a lot of them. About one every second or so. The connectivity kit also seems to send periodic packages back to the Prime. I think they are some kind of status check / ping between the prime and the connectivity kit.